doc/guide/build_windows.hpp

```diff
@@ -91,67 +85,81 @@ This tutorial follows the second approach for simplicity.
+- Install or unzip to "C:\boost\"
 
 <b> Armadillo Dependency </b>
 
 - Download "Armadillo" (armadillo-8.500.1.tar.xz) from <a href="http://arma.sourceforge.net/download.html">Sourceforge</a>
-- Unzip to "C:\mlpack\armadillo-8.500.1"
-- Create a "build" directory into "C:\mlpack\armadillo-8.500.1\"
-- Open the Command Prompt and navigate to "C:\mlpack\armadillo-8.500.1\build"
+- Unzip to "C:\mlpack\armadillo"
+- Create a "build" directory into "C:\mlpack\armadillo\"
+- Open the Command Prompt and navigate to "C:\mlpack\armadillo\build"
 - Run cmake: 
 
 @code
-cmake -G "Visual Studio 15 2017 Win64" -DBLAS_LIBRARY:FILEPATH="C:/mlpack/mlpack-3.0.4/packages/OpenBLAS.0.2.14.1/lib/native/lib/x64/libopenblas.dll.a" -DLAPACK_LIBRARY:FILEPATH="C:/mlpack/mlpack-3.0.4/packages/OpenBLAS.0.2.14.1/lib/native/lib/x64/libopenblas.dll.a" -DCMAKE_PREFIX:FILEPATH="C:/mlpack/armadillo" ..
+cmake -G "Visual Studio 15 2017 Win64" -DBLAS_LIBRARY:FILEPATH="C:/mlpack/mlpack/packages/OpenBLAS.0.2.14.1/lib/native/lib/x64/libopenblas.dll.a" -DLAPACK_LIBRARY:FILEPATH="C:/mlpack/mlpack/packages/OpenBLAS.0.2.14.1/lib/native/lib/x64/libopenblas.dll.a" ..
 @endcode
 
 @note If you are using different directory paths, a different configuration (e.g. Release)
 or a different VS version, update the cmake command accordingly.
 
-- Once it has successfully finished, open "C:\mlpack\armadillo-8.500.1\build\armadillo.sln"
+- Once it has successfully finished, open "C:\mlpack\armadillo\build\armadillo.sln"
 - Build > Build Solution
 - Once it has successfully finished, close Visual Studio
 
 @section build_windows_mlpack Building mlpack
 
-- Create a "build" directory into "C:\mlpack\mlpack-3.0.4\"
-- Use either the CMake GUI or the CMake command line to configure Armadillo.
+- Create a "build" directory into "C:\mlpack\mlpack\"
+- You can generate the project using either cmake via command line or GUI. If you prefer to use GUI, refer to the \ref build_windows_appendix "appendix"
+- To use the CMake command line prompt, open the Command Prompt and navigate to "C:\mlpack\mlpack\build"
+- Run cmake:
+
+@code
+cmake -G "Visual Studio 15 2017 Win64" -DBLAS_LIBRARY:FILEPATH="C:/mlpack/mlpack/packages/OpenBLAS.0.2.14.1/lib/native/lib/x64/libopenblas.dll.a" -DLAPACK_LIBRARY:FILEPATH="C:/mlpack/mlpack/packages/OpenBLAS.0.2.14.1/lib/native/lib/x64/libopenblas.dll.a" -DARMADILLO_INCLUDE_DIR="C:/mlpack/armadillo/include" -DARMADILLO_LIBRARY:FILEPATH="C:/mlpack/armadillo/build/Debug/armadillo.lib" -DBOOST_INCLUDEDIR:PATH="C:/boost/" -DBOOST_LIBRARYDIR:PATH="C:/boost/lib64-msvc-14.1" -DDEBUG=OFF -DPROFILE=OFF ..
+@endcode
+
+@note cmake will attempt to automatically download the ENSMALLEN dependency. If for some reason cmake can't download the dependency, you will need to manually download ENSMALLEN from http://ensmallen.org/ and extract it to "C:\mlpack\mlpack\deps\". Then, specify the path to ENSMALLEN using the flag: -DENSMALLEN_INCLUDE_DIR=C:/mlpack/mlpack/deps/ensmallen/include
```

Comment:
Is it intentional to write ensmallen in capital letters? When I talk about ensmallen I use the lower case from. My intention is to avoid the same situation that we see with mlpack which is mentioned in different flavors MLpack, MLPACK, mlpack, etc. I can see where MLPACK is coming from but I don't think MLpack is used in the codebase.

Reply:
Nice catch. I guess it's just a mind trick. I got used to LAPACK, and mlpack is defined as "a machine learning analog to LAPACK", therefore the extrapolation. And ensmallen, well the only reference I found in mlpack's Readme.MD is the cmake flag which is uppercase. My bad. Will fix it soon. Thanks!